test/functional/editor/ftplugin_spec.lua

```diff
@@ -0,0 +1,109 @@
+local t = require('test.testutil')
+local n = require('test.functional.testnvim')()
+
+local exec_lua = n.exec_lua
+local eq = t.eq
+local root = n.exec_lua('return vim.fs.normalize(vim.fn.getcwd())')
+
+before_each(function()
+  n.command [[
+    edit `=stdpath('config') .. '/lua/user-foo/init.lua'`
```

Comment:
this will only work if `clear()` was previously called in some other test. 

while we're here, I suggest moving this into the describe() block. that will be needed in the future anyway, if/when other ftplugin-related tests are added to this file.

```suggestion
setup(n.clear)
before_each(function()
  n.command [[
    edit `=stdpath('config') .. '/lua/user-foo/init.lua'`
```